packages/server-wallet/e2e-test/receiver/controller.ts

```diff
@@ -25,18 +24,23 @@ export default class ReceiverController {
   }
 
   public async acceptMessageAndReturnReplies(message: Payload): Promise<Payload> {
-    const {signedStates, objectives} = message;
+    const reply: Payload = {
+      signedStates: [],
+      objectives: [],
+    };
 
     const {
       channelResults: [channelResult],
+      outbox: [maybeSyncStateResponse],
     } = await this.time('push message', async () => this.wallet.pushMessage(message));
 
-    if (!signedStates?.length && !objectives?.length) {
-      return {
-        signedStates: [],
-        objectives: [],
-      };
-    } else {
+    if (maybeSyncStateResponse) {
+      const syncResponse = (maybeSyncStateResponse.params as WireMessage).data as Payload;
+      reply.signedStates = reply.signedStates?.concat(syncResponse.signedStates || []);
+      reply.objectives = reply.objectives?.concat(syncResponse.objectives || []);
+    }
+
+    if (channelResult && channelResult.turnNum % 2 === 0) {
```

Comment:
I don't follow the need for this condition.

Reply:
If the channel result had an odd turn number, then it would mean that it is _not_ the turn of the receiver, which would mean that the receiver should not try to update the channel (it would see a "not your turn" error). The case where it might be an odd turn number is introduced in this PR and would be the result of a channel syncing done by the payer where the payer tries to sync after having ignored the previous reply from the receiver (i.e., the 'can update pre-existing channel, send state, but ignore reply, and sync later' case).